packages/govuk-frontend/src/govuk/components/task-list/task-list.yaml

```diff
@@ -0,0 +1,250 @@
+    type: string
+    required: false
+    description: String to prefix ID for the tag and hint for each task list item. If `idPrefix` is not passed, fallback to using the `task-list` string instead.
+
+examples:
+  - name: default
+    data:
+      idPrefix: 'task-list-example'
+      items:
+        - title:
+            text: Company Directors
+          href: '#'
+          status:
+            text: Completed
+            classes: govuk-tag--black
+
+        - title:
+            text: Registered company details
+          href: '#'
+          status:
+            tag:
+              text: Incomplete
+              classes: govuk-tag--blue
+
+        - title:
+            text: Business plan
+          href: '#'
+          status:
+            tag:
+              text: Incomplete
+              classes: govuk-tag--blue
+
+  - name: example with 3 states
+    data:
+      idPrefix: 'task-list-example'
+      items:
+        - title:
+            text: Company Directors
+          href: '#'
+          status:
+            text: Completed
+        - title:
+            text: Registered company details
+          href: '#'
+          status:
+            tag:
+              text: Not started
+              classes: govuk-tag--light-blue
+        - title:
+            text: Business plan
+          href: '#'
+          status:
+            tag:
+              text: In progress
+              classes: govuk-tag--blue
+        - title:
+            text: Documentation
+          href: '#'
+          status:
+            tag:
+              text: Not started
+              classes: govuk-tag--light-blue
+
+  - name: example with hint text and additional states
+    data:
+      idPrefix: 'task-list-example'
+      items:
+        - title:
+            text: Company Directors
+          href: '#'
+          status:
+            text: Completed
+        - title:
+            text: Registered company details
+          href: '#'
+          status:
+            tag:
+              text: Not started
+              classes: govuk-tag--light-blue
+        - title:
+            text: Business plan
+          href: '#'
+          hint:
+            text: Ensure the plan covers objectives, strategies, sales, marketing and financial forecasts.
+          status:
+            tag:
+              text: Review
+              classes: govuk-tag--pink
+        - title:
+            text: Documentation
+          href: '#'
+          status:
+            tag:
+              text: In progress
+              classes: govuk-tag--blue
+        - title:
+            text: Charitable status
+          href: '#'
+          status:
+            tag:
+              text: Error
+              classes: govuk-tag--red
+        - classes: app-task-list__item--no-link
+          title:
+            text: Payment
+          hint:
+            text: It will cost between £15 and £75
+          status:
+            text: Cannot start yet
+            classes: govuk-hint
```

Comment:
Don't use a class from another component that's not being imported, because:

1. there's no guarantee that the user will be including the styles for the hint component in the future
2. we might change the styles for the hint component in the future (for example, adding a margin, or making other changes) that would affect this component as well.

Suggest instead adding a modifier class:

```scss
.govuk-task-list__status--muted { // or similar, maybe .govuk-task-list__status--secondary
  color: $govuk-secondary-text-colour;
}
```

Reply:
Good point! I think the `govuk-hint` class is widely abused for non-field-hints (we might even do that in production, opps), but we should defo avoid it here.

How about `--cannot-start-yet` as the modifier, if we want to be more prescriptive? I don’t think we've uncovered any other use-cases for grey text in the status?

Reply:
Added now, thanks!